File: tests/fixtures/sample.py
"""Sample Python module for testing."""


def add(x: int, y: int) -> int:
    """Add two numbers."""
    return x + y


def decorator(func):
    """A simple decorator."""
    return func


@decorator
def decorated_function() -> None:
    """A decorated function."""
    pass


class Calculator:
    """A simple calculator."""

    def multiply(self, x: int, y: int) -> int:
        """Multiply two numbers."""
        return x * y

    def divide(self, x: float, y: float) -> float:
        return x / y


class Outer:
    """An outer class with a nested class."""

    class Inner:
        """A nested inner class."""

        def inner_method(self) -> str:
            return "inner"


# Module-level constant — should NOT be extracted as a symbol
MY_CONSTANT = 42

File: tests/fixtures/sample.rs
//! Sample Rust module for testing.

/// Adds two integers together.
pub fn add(x: i32, y: i32) -> i32 {
    x + y
}

/// A simple counter struct.
pub struct Counter {
    count: u32,
}

impl Counter {
    /// Creates a new Counter starting at zero.
    pub fn new() -> Self {
        Counter { count: 0 }
    }

    /// Increments the counter by one.
    pub fn increment(&mut self) {
        self.count += 1;
    }

    /// Returns the current count.
    pub fn value(&self) -> u32 {
        self.count
    }
}

/// A trait for types that can describe themselves.
pub trait Describable {
    fn describe(&self) -> String;
}

/// Primary color variants.
pub enum Color {
    Red,
    Green,
    Blue,
}

// MAX_COUNT is a constant — should NOT be extracted as a symbol.
const MAX_COUNT: u32 = 100;
